### ISSUE TITLE
document how to integrate into WSGI servers

### ISSUE THREAD

**Issue by faassen** [morepath/morepath#205]
To integrate with an arbitrary WSGI server, we need to explain in the docs how to create a module that isn't imported by anything else that imports the App, instantiates it (so that you can point a WSGI server to it), and also does a configuration scan and commit on the module-level.

This because WSGI servers don't all allow configuration with a factory function that returns the WSGI object; many just want the WSGI object straight away, but Morepath needs to finish configuration before that is ready, and this is a way to accomplish this.

**Comment by faassen** [comment]
Thanks to work by @taschini this now starts to look pretty simple.

**Comment by faassen** [comment]
There are different types of WSGI server:

* ones you start from Python code (say, waitress). Here you can do an explicit commit before you start serving requests.

* ones like Apache and nginx. Here they import some Python module. Do we want to do the autoscan and commit in a ``if __name__ == '__main__'`` block? Is the name even ``__main__`` when integrating with Apache and nginx?

This impacts the discussion in #392

**Comment by faassen** [comment]
Hm, the documentation for other Python web servers that integrate with nginx all seem to use some kind of proxy pattern where they run a Python web server behind an nginx frontend. In which case you always do the configuration in python code.

**Comment by faassen** [comment]
Relevant: http://flask.pocoo.org/docs/0.10/deploying/mod_wsgi/

**Comment by href** [comment]
AFAIK nginx doesn't have builtin Python support. However it *has* builtin uWSGI support (which has it's own C API for performance). Personally I'm usually just proxying to gunicorn or chausette.

**Comment by faassen** [comment]
Yes, that seems to be the common nginx setup, and in this case you're in control over the backend server in Python and can do an explicit commit. In Apache mod_wsgi you can integrate directly (see the flask doc), but there you have a wsgi file where you can do explicit Python stuff.
